cmd/otel-allocator/allocation/allocator.go

```diff
@@ -47,103 +58,96 @@ func (t TargetItem) hash() string {
 
 // Create a struct that holds collector - and jobs for that collector
 // This struct will be parsed into endpoint with collector and jobs info
-
+// This struct can be extended with information like annotations and labels in the future
 type collector struct {
 	Name       string
 	NumTargets int
 }
 
+type State struct {
+	// collectors is a map from a collector's name to a collector instance
+	collectors map[string]collector
+	// targetItems is a map from a target item's hash to the target items allocated state
+	targetItems map[string]TargetItem
+}
+
+func NewState(collectors map[string]collector, targetItems map[string]TargetItem) State {
```

Comment:
Why should this function be exposed?

Reply:
It can't be used outside of this package as it is since it requires an unexported type as one of its arguments.